api/v1alpha1/openapi.yaml

```diff
@@ -2549,26 +2521,131 @@ components:
+    GitRepositoryRef:
+      type: object
+      properties:
+        repository:
+          type: string
+          description: |
+            The name of the repository resource to use as the sync source
+        targetRevision:
+          type: string
+        path:
+          type: string
+        mountPath:
+          description: Path to config in device
+          type: string
+          default: "/"
+      required:
+        - repository
+        - targetRevision
+        - path
+    GitEnvRepositoryRef:
+      type: object
+      properties:
+        repository:
+          type: string
+          description: |
+            The name of the repository resource to use as the sync source for environment variables
+        targetRevision:
+          type: string
+          description: |
+            The specific revision of the repository to pull from
+        envFilePath:
+          type: string
+          description: |
+            Path to a file within the repository that contains environment variables to be consumed.
+      required:
+        - repository
+        - targetRevision
+        - envFilePath
+    KubernetesSecretRepositoryRef:
+      type: object
+      properties:
+        name:
+          type: string
+        namespace:
+          type: string
+        mountPath:
+          type: string
+      required:
+        - name
+        - namespace
+        - mountPath
+    HttpRepositoryRef:
+      type: object
+      properties:
+        repository:
+          type: string
+          description: |
+            The name of the repository resource to use as the sync source
+        suffix:
+          type: string
+          description: |
+            Part of the URL that comes after the base URL. It can include query parameters such as:
+            /path/to/endpoint?query=param
+        filePath:
+          type: string
+          description: |
+            The path of the file where the response is stored in the filesystem of the device.
+      required:
+        - repository
+        - filePath
+    ApplicationSpec:
+      type: object
+      description: "Defines the specification for an application"
+      properties:
+        name:
+          type: string
+          description: "The unique name of the application"
+        bundle:
+          $ref: '#/components/schemas/ApplicationBundleProviderSpec'
+        envVars:
+          $ref: '#/components/schemas/ApplicationEnvVars'
+        envFiles:
+          type: array
+          description: "Environment variable key-value pairs from env files, injected at runtime."
+          items:
+            $ref: '#/components/schemas/GitEnvRepositoryRef'
+      required:
+        - name
+        - bundle
+    ApplicationEnvVars:
+      type: object
+      description: "Environment variable key-value pairs, injected during runtime"
+      additionalProperties:
+        type: string
+    ApplicationBundleImageRef:
+      type: object
+      properties:
+        image:
```

Comment:
I think we could consider moving this back to a string I extended to an object to allow for concerning things like authentication. But I believe bundle can  have `secrets:`  which could provide this auth